pkg/vpcmodel/vpcConnectivity.go

```diff
@@ -73,3 +79,43 @@ func NewConfigBasedConnectivityResults() *ConfigBasedConnectivityResults {
 		EgressAllowedConns:  map[string]*common.ConnectionSet{},
 	}
 }
+
+func (v *VPCConnectivity) SplitAllowedConnsToUnidirectionalAndBidirectional() (
+	bidirectional, unidirectional NodesConnectionsMap) {
+	unidirectional = NewNodesConnectionsMap()
+	bidirectional = NewNodesConnectionsMap()
+	for src, connsMap := range v.AllowedConnsCombined {
+		for dst, conn := range connsMap {
+			if conn.IsEmpty() {
+				continue
+			}
+			statefulConn := v.AllowedConnsCombinedStateful.getAllowedConnForPair(src, dst)
+			switch {
+			case conn.Equal(statefulConn):
+				bidirectional.updateAllowedConnsMap(src, dst, conn)
+			case statefulConn.IsEmpty():
+				unidirectional.updateAllowedConnsMap(src, dst, conn)
+			default:
+				bidirectional.updateAllowedConnsMap(src, dst, statefulConn)
+				unidirectional.updateAllowedConnsMap(src, dst, conn.Subtract(statefulConn))
+			}
+		}
+	}
+	return bidirectional, unidirectional
+}
+
+func (nodesConnMap *NodesConnectionsMap) updateAllowedConnsMap(src, dst Node, conn *common.ConnectionSet) {
```

Comment:
```suggestion
func (nodesConnMap NodesConnectionsMap) updateAllowedConnsMap(src, dst Node, conn *common.ConnectionSet) {
```
In Go, maps hold references to an underlying data structure. If you pass a map to a function that changes the contents of the map, the changes will be visible in the caller. 

Please make this change in the other two `NodesConnectionsMap` methods as well.

Reply:
right.. fixed